### PR TITLE
feat: show web vitals event properties in player inspector list

### PR DESCRIPTION
**Pull request by pauldambra**
* premptively load web vitals data in replay so its available for display
* show it alongside other performance summary data

Screenshot 2024-06-29 at 13 32 57
Screenshot 2024-06-30 at 20 46 52
Screenshot 2024-06-30 at 20 46 59
Screenshot 2024-06-30 at 20 47 11

### CODE REVIEW

File: frontend/src/scenes/session-recordings/apm/performanceEventDataLogic.ts [hunk 1]
```diff
@@ -24,14 +24,19 @@ export interface PerformanceEventDataLogicProps extends SessionRecordingDataLogi
...
+    return events.sort((a, b) => (a.timestamp.valueOf() > b.timestamp.valueOf() ? 1 : -1))
+}
+
 /**
  * If we have paint events we should add them to the appropriate navigation event
  * this makes it easier to draw performance cards for navigation events
  */
 function matchPaintEvents(performanceEvents: PerformanceEvent[]): PerformanceEvent[] {
-    // KLUDGE: this assumes that the input is sorted by timestamp and relies on the identity of the events to mutate them
+    // NB: this relies on the input being sorted by timestamp and relies on the identity of the events to mutate them
```
Comment (marandaneto) [inline frontend/src/scenes/session-recordings/apm/performanceEventDataLogic.ts:37]: What NB means?

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts [hunk 2]
```diff
@@ -662,6 +676,12 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
         },
     })),
     selectors(({ cache }) => ({
+        webVitalsEvents: [
+            (s) => [s.sessionEventsData],
+            (sessionEventsData): RecordingEventType[] =>
+                (sessionEventsData || []).filter((e) => e.event === '$web_vitals'),
```
Comment (marandaneto) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:682]: I'd love to start using constants for such strings, we have so many across our code that replacing or making typos is super easy.
says the person used to strongly typed languages :P

**Review by marandaneto — APPROVED** [review]

File: frontend/src/scenes/session-recordings/apm/performanceEventDataLogic.ts (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/apm/performanceEventDataLogic.ts:37]: Ah... it's latin "Nota bene"

https://en.wikipedia.org/wiki/Nota_bene

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts (hunk 2 above)
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:682]: tbh we could have a type in the front end that expresses all the known event names + and string so you don't need to declare constants and you get type safety 🤔

File: frontend/src/scenes/session-recordings/apm/performanceEventDataLogic.ts (hunk 1 above)
Comment (marandaneto) [reply frontend/src/scenes/session-recordings/apm/performanceEventDataLogic.ts:37]: TIL

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts (hunk 2 above)
Comment (marandaneto) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:682]: That sounds like a nice refactoring idea for the future Paul, not a blocker (it's merged anyway :D)

**Pull request merged**